src/accelerate/accelerator.py

```diff
@@ -1415,35 +1415,36 @@ def _prepare_deepspeed(self, *args):
         deepspeed_plugin = self.state.deepspeed_plugin
 
         is_dataloader_present = any(isinstance(obj, torch.utils.data.DataLoader) for obj in args)
-        if deepspeed_plugin.deepspeed_config["train_micro_batch_size_per_gpu"] == "auto" or is_dataloader_present:
-            result = [
-                self._prepare_one(obj, first_pass=True) if isinstance(obj, torch.utils.data.DataLoader) else obj
-                for obj in args
-            ]
-
-            batch_sizes = [obj.batch_size for obj in args if hasattr(obj, "batch_size")]
-            if self.split_batches:
-                batch_sizes = [batch_size // self.num_processes for batch_size in batch_sizes]
+        if deepspeed_plugin.deepspeed_config["train_micro_batch_size_per_gpu"] == "auto":
+            if is_dataloader_present:
+                result = [
+                    self._prepare_one(obj, first_pass=True) if isinstance(obj, torch.utils.data.DataLoader) else obj
+                    for obj in args
+                ]
```

Comment:
Changes:
1. the indentation was wrong in your original code
2. moved logic around to deal with separate needs separately

Reply:
Also since `result` is being used much later from where it's created, perhaps renaming this var to something more sensible would make the code more readable? Just a suggestion.